lib/functions/manga.js

```diff
@@ -0,0 +1,95 @@
+"use strict";
+
+var common = require("./common");
+
+var setPropatiesToNextPage = function(props){
+    var div = document.getElementsByClassName("works_display")[0];
+    var a = div.children[0];
+    a.href = a.href + "&namingdata=" + encodeURIComponent(JSON.stringify(props));
+};
+
+var getPropatiesFromPreviousPage = function(){
+    var url = document.location.href,
+    paramIndex = url.indexOf("?"),
+    params = url.substr(paramIndex + 1).split("&");
+    var res;
+    params.forEach(function(param){
+        var tmp = param.split("="),
+        key = tmp[0],
+        value = tmp[1];
+        if(key === "namingdata"){
+            res = value;
+            return;
+        }
+    });
+    return JSON.parse(decodeURIComponent(res));
+};
+
+var setDownloadLink = function(type, props, pageNum){
+    if (type === "manga") {
+        props.page = pageNum;
+    }
+
+    var img = document.evaluate("/html/body/img", document, null, XPathResult.FIRST_ORDERED_NODE_TYPE, null ).singleNodeValue;
+    var url = getillustUrlFromImage(img);
```

Comment:
__[eslint]__
*  Error - getillustUrlFromImage was used before it was defined (no-use-before-define)